tests/golden/check-local-references-outside-repo/check-local-references-outside-repo.bats

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bats
+
+# SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+load '../helpers/bats-support/load'
+load '../helpers/bats-assert/load'
+load '../helpers'
+
+@test "We report all links that target files outside root" {
+  xrefcheck -r inner-directory\
+  | prepare > /tmp/check-notScanned.test || true
+
+  diff /tmp/check-notScanned.test expected.gold \
+    --ignore-space-change \
+    --ignore-blank-lines \
+    --new-file # treat absent files as empty
+
+  rm /tmp/check-notScanned.test
+}
+
+@test "When target is inside repo root, everything is good" {
+  run xrefcheck -r .\
+
+  assert_output --partial "All repository links are valid."
+}
+
+@test "When target is a virtual file, everything is good" {
+  run xrefcheck -c config-virtual-file.yaml -r inner-directory\
+
+  assert_output --partial "All repository links are valid."
+}
```

Comment:
Great tests! Testing a scenario where the root directory is the same as the current directory (`--root .`) **and** also when it's different than the current directory (`--root inner-directory`) is a great idea.

Let's add a bit more coverage. We're currently testing "relative filepaths" (`../etc`), let's also test with "absolute paths" (`/../etc`).

Reply:
Done. During this I have discovered that virtual files in config must be "relative", idk is that expected behavior, maybe we should warn users when see `/../a.md` in `virtualFiles`.

Reply:
> Done

Great! One last thing, let's also test this scenario from the description of #138: 

![image](https://user-images.githubusercontent.com/718166/193022116-997f8e8b-eb75-40a9-86ca-1c99f8e850c8.png)


For example, say the file `inner-directory/check-local-references-outside-repo.md` contains this link:

```md
[](../inner-directory/check-local-references-outside-repo.md)
```

And we run `xrefcheck -r inner-directory`. Technically, the link points to a file inside the repo root - but _part of it_ doesn't. While resolving the path, we'd have to step outside the repo root and then back in, which makes the link illegal.

> During this I have discovered that virtual files in config must be "relative", idk is that expected behavior, maybe we should warn users when see /../a.md in virtualFiles.

Oh great catch! Let me look a bit more into this and I'll get back to you.

Reply:
I am currently working on #165 (making test folder  `test-local-refs`), and I think I could rebase this over my current branch and make this check a part of big tests family

Reply:
> Done. During this I have discovered that virtual files in config must be "relative", idk is that expected behavior, maybe we should warn users when see `/../a.md` in `virtualFiles`. 

Okay, I've had a chance to dig into this.

Absolute paths in `virtualFiles` (and anywhere else where glob patterns are expected, like `ignored` and `notScanned`) "works", but perhaps not in the way you expected.


* In general, absolute paths start from the **filesystem's** root.
* However, in a markdown file, github/gitlab interpret absolute paths as starting from the **repo's** root. This is the **only** context where absolute paths do not refer to the filesystem's root.

In that sense, when `virtualFiles/ignored/notScanned` is given an absolute path, it'll be interpreted as starting from the filesystem's root. E.g., here's an example that works on my machine:

```
$ cd /home/dc/dev/serokell/xrefcheck-test/virtualFiles-absolute-paths

$ echo '[](invalid)' > a.md

$ xrefcheck --no-progress --ignored /home/dc/dev/serokell/xrefcheck-test/virtualFiles-absolute-paths/a.md
All repository links are valid.
```

Or with `virtualFiles`:

```
$ echo "verification:\n  virtualFiles:\n    - /home/dc/dev/serokell/xrefcheck-test/virtualFiles-absolute-paths/invalid" > .xrefcheck.yaml

$ xrefcheck --no-progress
All repository links are valid.
```

Reply:
So, should we warn users or change this behavior? We should not depend on location of our root in filesystem anyway, imho if xrefcheck runs without errors once, it should run without errors for anyone who clone a repo

Reply:
Agreed - using absolute paths here hurts reproducibility.

Locally, supporting absolute paths might still be useful when copy/pasting filepaths around to quickly test something.

So do we want to forbid absolute paths flat out? Or support them but emit a warning?

Personally, I'm tempted to say it's not a big deal - let's go with the usual rule, let's keep it simple for now and just flat out forbid it.

Can you please create an issue for this?

Reply:
Ok

Reply:
> I could rebase this over my current branch and make this check a part of big tests family

Done